apps/site/src/app/admin/applicants/[uid]/components/HackerApplicantActions.tsx

```diff
@@ -0,0 +1,91 @@
+import { useContext, useState } from "react";
+
+import {
+	Box,
+	Button,
+	Input,
+	SpaceBetween,
+} from "@cloudscape-design/components";
+import {
+	Review,
+	submitReview,
+} from "@/app/admin/applicants/hackers/useApplicant";
+import { Uid } from "@/lib/userRecord";
+import UserContext from "@/lib/admin/UserContext";
+import { isReviewer } from "@/lib/admin/authorization";
+
+interface ColoredTextBoxProps {
+	text: string | undefined;
+}
+
+const ColoredTextBox = ({ text }: ColoredTextBoxProps) => {
+	return (
+		<Box variant="span" color="text-status-error" fontWeight="heavy">
+			{text}
+		</Box>
+	);
+};
+
+interface ApplicantActionsProps {
+	applicant: Uid;
+	reviews: Review[];
+	submitReview: submitReview;
+}
+
+function HackerApplicantActions({
+	applicant,
+	reviews,
+	submitReview,
+}: ApplicantActionsProps) {
+	const { uid, roles } = useContext(UserContext);
+	const [value, setValue] = useState("");
+
+	const uniqueReviewers = Array.from(
+		new Set(reviews.map((review) => review[1])),
+	);
+
+	// const canReview = either there are less than 2 reviewers or uid is in current reviews
+	const canReview = uid
+		? uniqueReviewers.length < 2 || uniqueReviewers.includes(uid)
+		: false;
+
+	if (!isReviewer(roles)) {
+		return null;
+	}
+
+	return canReview ? (
+		<SpaceBetween direction="horizontal" size="xs">
+			<Input
+				onChange={({ detail }) => setValue(detail.value)}
+				value={value}
+				type="number"
+				inputMode="decimal"
+				placeholder="Applicant score"
+				step={0.5}
+				disabled={!canReview}
+			/>
+			<Button
+				onClick={() => {
+					submitReview(applicant, value);
+					setValue("");
+				}}
+				disabled={!canReview}
+			>
+				Submit
+			</Button>
+		</SpaceBetween>
+	) : (
+		<SpaceBetween size="xxs">
+			<Box variant="awsui-key-label" color="text-status-info">
+				<ColoredTextBox text={uniqueReviewers[0].split(".").at(-1)} /> and{" "}
+				<ColoredTextBox text={uniqueReviewers[1].split(".").at(-1)} /> already
+				submitted reviews.
+			</Box>
+			<Box variant="awsui-key-label" color="text-status-info">
+				Log in as one of them to submit a review.
```

Comment:
Question: this direction seems rather puzzling since in practice, reviewers cannot impersonate each other. Is there a better way to indicate the course of action here?

Reply:
This can be changed to something like "Contact Rosalind, Nicole, and Albert if you think this is an error."